src/pages/lessons/fundamentals/eth-intro-part-2.mdx

```diff
@@ -0,0 +1,279 @@
+
+So, what does a transaction look like if not from an unlocked account? To find out, let's send some ether from `acct_two` – an account not managed by `EthereumTesterProvider`. This more manual process takes three steps: 1) specify the transaction details, 2) sign the transaction, then 3) broadcast the transaction to the network.
+
+```python
+# 1) manually build a transaction
+In [9]: tx = {
+          'to': acct_one,
+          'value': 10000000,
+          'gas': 21000,
+          'gasPrice': w3.eth.get_block('pending')['baseFeePerGas'],
+          'nonce': 0
+        }
+
+# 2) sign the transaction with the sender's private key
+In[10]: signed = w3.eth.account.sign_transaction(tx, acct_two.key)
+
+# 3) send the "raw" transaction
+In[11]: tx_hash = w3.eth.send_raw_transaction(signed.rawTransaction)
+```
+
+Let's break that down. Step 1 defines a Python dictionary with the required transaction fields. We briefly learned about gas (transaction fees) in Part 1, but nonce may be new territory. In Ethereum, a **nonce** is simply the transaction count of the account. The Ethereum protocol keeps track of this value to prevent double-spending.
+
+Since this is the first transaction being made by `acct_two`, its nonce is zero. If you supply the wrong value, the result is an invalid transaction and is rejected by web3.py:
+
+```bash
+ValidationError: Invalid transaction nonce: Expected 0, but got 4
+```
+
+Note that a nonce is still required when sending a transaction from `acct_one`, but `EthereumTesterProvider` keeps track of transaction counts for managed accounts and adds the appropriate nonce to new transactions.
+
+Another detail you may have noticed is that a `from` value is missing from `tx`. In this case, the `sign_transaction` method can infer the sender's address from their private key. Again, a public address can be derived from a private key, but a private key cannot be reverse engineered from its public address.
+
+Finally, a "raw" transaction is just the transaction data and signature represented as bytes. Under the hood, `send_transaction` performs the same encoding required by `send_raw_transaction`.
+
+### Deploying smart contracts
+
+A full introduction to smart contracts will be covered in the next lesson, but this is a good opportunity for a sneak peak; interacting with smart contracts looks very similar to a standard transaction.
+
+Briefly, **smart contracts** are programs that live on the Ethereum blockchain, available for anyone to use. When you're ready to deploy a smart contract, you compile the code down to bytecode and include it in a transaction as a `data` value:
+
+```python
+bytecode = "6080604052348015610...36f6c63430006010033"
+
+tx = {
+   'data': bytecode,
+   'value': 0,
+   'nonce': 0,
+   ...
+}
+```
+
+Besides requiring more gas, the only other distinction in a contract deployment transaction is the absence of a `to` value. The rest of the process is identical to a standard transfer of ether.
+
```

Comment:
Is there no `to` value because the smart contract's deployment creates the address it's going to go `to`/land on?

Reply:
Shmoooooothly done!